.github/workflows/regression-tests.yml

```
@@ -0,0 +1,84 @@
+name: Regression Tests
+'on':
+  workflow_dispatch:
+    inputs:
+      zkevm_agglayer_commit_id:
+        description: 0xPolygon/agglayer (commit id)
+        required: true
+      bake_time:
+        description: bake time (minutes)
+        required: false
+        default: 30
+jobs:
+  deploy_devnet:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: '${{ github.event.before }}'
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v1
+      - name: Clone and build agglayer image locally
+        run: |
+          git clone https://github.com/0xPolygon/agglayer.git
+          cd agglayer
+          git checkout "${{ github.event.inputs.zkevm_agglayer_commit_id }}"
+          docker compose -f docker/docker-compose.yaml build --no-cache agglayer
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+      - name: Clone internal kurtosis-cdk repo
+        run: |
+          git clone https://github.com/0xPolygon/kurtosis-cdk.git
+          cd kurtosis-cdk
+          git checkout dan/jit_containers
+      - name: Install kurtosis
+        run: |
+          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
+          sudo apt update
+          sudo apt install kurtosis-cli
```

nit: You can disable `kurtosis analytics` when running the tool in CI

```suggestion
          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
          sudo apt update
          sudo apt install kurtosis-cli
          kurtosis analytics disable
```